src/store/actions/user.actions.ts

```diff
@@ -56,17 +66,31 @@ export const addIdentifier = (args: IAddIdentifierArgs): ThunkAction<Promise<voi
 };
 
 export const setActiveUser = (userId: string): ThunkAction<Promise<void>, RootState, unknown, Action> => {
-  return async (dispatch: ThunkDispatch<RootState, unknown, Action>) => {
+  return async (dispatch: ThunkDispatch<RootState, unknown, Action>, getState: CombinedState<any>) => {
     dispatch({type: USERS_LOADING});
-    getUsersFromStorage()
-      .then((users: Map<string, IUser>) => {
+    await getUsersFromStorage()
+      .then(async (users: Map<string, IUser>) => {
         const user = users.get(userId);
         if (user) {
           dispatch({type: SET_ACTIVE_USER_SUCCESS, payload: user});
+          let userState: IUserState = getState().user;
+          while (!userState.activeUser) {
+            await new Promise((resolve) => setTimeout(resolve, 50));
+            userState = getState().user;
+          }
+          await dispatch(getContacts())
+          let contactState: IContactState = getState().contact;
+          while (contactState.contacts.length === 0) {
```

Comment:
Please add a remark that the only reason this works is because we always will have at least 1 contact (the user)

Reply:
added a comment